Content.Server/Power/SMES/PowerSmesSystem.cs

```diff
@@ -1,16 +1,75 @@
+using Content.Server.Power.Components;
+using Content.Server.Power.EntitySystems;
+using Content.Shared.Power;
+using Content.Shared.Rounding;
+using Content.Shared.SMES;
 using JetBrains.Annotations;
+using Robust.Shared.Timing;
 
 namespace Content.Server.Power.SMES
 {
     [UsedImplicitly]
     internal sealed class PowerSmesSystem : EntitySystem
     {
-        public override void Update(float frameTime)
```

Comment:
you still need the update implementation here to match how PowerSmesSystem updates SMES components beforehand

Reply:
> you still need the update implementation here to match how PowerSmesSystem updates SMES components beforehand

Why is it not sufficient to use MapInit and ChargeChange events the way ApcSystem does? https://github.com/space-wizards/space-station-14/blob/6f1eac34d5036ae9938a24cd2f6eed39a0eb8f75/Content.Server/Power/EntitySystems/ApcSystem.cs#L45

(New to codebase, sorry if I'm missing something here)

Reply:
yeah, that's a good point - it looks like the old update method was just a quick hack